app/controllers/order/client_information.js

```diff
@@ -72,19 +87,60 @@ export default Ember.Controller.extend(cancelOrder, {
         actionType = "POST";
       }
 
-      var loadingView = getOwner(this).lookup('component:loading').append();
+      let loadingView = getOwner(this).lookup('component:loading').append();
 
-      new AjaxPromise(url, actionType, this.get('session.authToken'), { beneficiary: this.beneficiaryParams(), order_id: orderId })
+      if (clientInfo === 'organisation') {
+        let orderParams = {
+          'purpose_ids': purposeIds,
+          'beneficiary_id': null
+        };
+
+        new AjaxPromise('/orders/' + orderId, 'PUT', this.get('session.authToken'), { order: orderParams })
         .then(data => {
-          this.get("store").pushPayload(data);
-          loadingView.destroy();
-          if(this.get("previousRouteName") === "my_orders") {
-            this.get("myOrders").set("selectedOrder", this.get("order"));
-            this.transitionToRoute('my_orders');
-          } else {
-            this.transitionToRoute('order.goods_details', orderId, { queryParams: { fromClientInformation: true }});
+          this.store.pushPayload(data);
+          if (beneficiaryId) {
+            let beneficiary = this.store.peekRecord('beneficiary', beneficiaryId);
+            if(beneficiary) {
+              new AjaxPromise("/beneficiaries/" + beneficiaryId, 'DELETE', this.get('session.authToken'))
+              .then(() => {
+                this.store.unloadRecord(beneficiary);
+                loadingView.destroy();
+              });
+            }
           }
+          this.send('redirectToGoodsDetails');
         });
+      } else {
+        let orderParams = {
+          'purpose_ids': purposeIds
+        };
+
+        new AjaxPromise('/orders/' + orderId, 'PUT', this.get('session.authToken'), { order: orderParams })
+        .then(() => {
+          new AjaxPromise(url, actionType, this.get('session.authToken'), { beneficiary: this.beneficiaryParams(), order_id: orderId })
+          .then(data => {
+            this.get("store").pushPayload(data);
+            loadingView.destroy();
```

Comment:
It's not a good practice to remove loadingview in **then** block. If there is any error then it won't go in catch and finally. 

So I would advise you to add catch and finally block and make changes accordingly.